dev/com.ibm.ws.wsoc/src/com/ibm/ws/wsoc/SessionImpl.java

```diff
@@ -137,10 +137,13 @@ public void signalAppOnOpen(WsByteBuffer remainingBuf, boolean runWithDoPriv) {
 
         connLink.setEndpointManager(things.getEndpointManager());
 
-        // Not sure if we should do this right before or right after onOpen, spec is ambigioius, seems like right before is better.
-        userProperties = new HashMap<String, Object>();
-        things.setUserProperties(userProperties);
-
+        if(WebSocketVersionServiceManager.isWsoc21rHigher()){
```

Comment:
Please open an issue so we don't forget to fix this to be `isWsoc21orHigher`

Reply:
https://github.com/OpenLiberty/open-liberty/issues/22257